mapper/orm/src/main/java/org/hibernate/search/mapper/orm/massindexing/impl/IdentifierConsumerDocumentProducer.java

```diff
@@ -190,37 +191,38 @@ private void rollbackTransaction(SessionImplementor session) throws Exception {
 		}
 	}
 
-	private void indexAllQueue(List<?> entities) throws InterruptedException, ExecutionException {
+	private void indexAllQueue(SessionImplementor session, List<?> entities) throws InterruptedException, ExecutionException {
 		if ( entities == null || entities.isEmpty() ) {
 			return;
 		}
-		else {
-			log.tracef( "received a list of objects to index: %s", entities );
-			for ( Object object : entities ) {
-				try {
-					index( object );
-					// TODO: implements monitor
-					// monitor.documentsBuilt( 1 );
-				}
-				catch (RuntimeException e) {
-					String errorMsg = log.massIndexerUnableToIndexInstance(
-							object.getClass().getName(),
-							object.toString()
-					);
-
-					// TODO: implements exception handler
-					// errorHandler.handleException( errorMsg, e );
-					// temporary re-throw the exception
-					throw new RuntimeException( errorMsg, e );
-				}
+
+		//TODO: switch to specific batch-optimized workPlan
+		PojoWorkPlan workPlan = mapping.createSearchManager( session ).createWorkPlan();
+
+		log.tracef( "received a list of objects to index: %s", entities );
+		for ( Object object : entities ) {
+			try {
+				index( workPlan, object );
+				// TODO: implements monitor
+				// monitor.documentsBuilt( 1 );
+			}
+			catch (RuntimeException e) {
```

Comment:
Note for later, when you implement the other "work plan": we really need to process the entity in this `index()` method. Currently that task is delayed to the execute() call below, so we're unlikely to ever get an exception here.

Reply:
Sure